source/labels/labels.go

```diff
@@ -65,26 +68,40 @@ func (c *labelCollector) Execute(ctx context.Context, _ context.CancelFunc) erro
 		c.logger.WithError(err).Error("failed to query metrics")
 		return nil
 	}
-	result := make(map[string][]string)
+
+	var (
+		mtx    sync.Mutex
+		ch     = make(chan struct{}, c.concurrency)
+		result = make(map[string][]string, len(labelValues))
+	)
 	for _, metricName := range labelValues {
-		c.logger.Debugf("querying Prometheus to get label names for metric %s", metricName)
-		labels, _, queryErr := c.promClient.LabelNames(ctx, []string{string(metricName)}, start, now)
-		if queryErr != nil {
-			c.logger.WithError(queryErr).Errorf("failed to query labels for the metric %q", metricName)
-			continue
-		}
-		result[string(metricName)] = removeLabelName(labels)
-	}
-	if len(result) > 0 {
-		if c.metricUsageClient != nil {
-			// In this case, that means we have to send the data to a remote server.
-			if sendErr := c.metricUsageClient.Labels(result); sendErr != nil {
-				c.logger.WithError(sendErr).Error("Failed to send labels name")
+		ch <- struct{}{}
```

Comment:
I think if you use the package `perses/common/async` it will simplify the whole process https://github.com/perses/common/blob/main/async/async.go#L14-L32

Here an example on how you can use it (in addition of the documentation) https://github.com/perses/plugins/blob/main/scripts/build-plugins/build-plugins.go

Reply:
After looking at the Async/Await pattern, I fail to see how I can use it directly.
In the case here, we want to run a bunch of tasks concurrently but throttle the number of active tasks. Is it something that should be added to the async package?